numpyro/contrib/autoguide/__init__.py

```diff
@@ -203,44 +185,16 @@ def __call__(self, *args, **kwargs):
             site = self.prototype_trace[name]
             value = transform(unconstrained_value)
             log_density = - transform.log_abs_det_jacobian(unconstrained_value, value)
-            if site['intermediates']:
-                event_ndim = len(site['fn'].base_dist.event_shape)
-            else:
-                event_ndim = len(site['fn'].event_shape)
+            event_ndim = len(site['fn'].event_shape)
             log_density = sum_rightmost(log_density,
                                         np.ndim(log_density) - np.ndim(value) + event_ndim)
             delta_dist = dist.Delta(value, log_density=log_density, event_ndim=event_ndim)
             result[name] = numpyro.sample(name, delta_dist)
 
         return result
 
-    def _unpack_and_constrain(self, latent_sample, params):
-        sample_shape = np.shape(latent_sample)[:-1]
-        # XXX: we do not support priors with supports depending on dynamic data
-        # because it adds complexity to the interface.
-        # Users can achieve that behaviour by changing the default `self._args`
-        # but we will not recommend doing so.
-        model_args = self._args
-        model_kwargs = self._kwargs
-
-        def unpack_single_latent(latent):
-            unpacked_samples = self._unpack_latent(latent)
-            if self._has_transformed_dist:
-                # first, substitute to `param` statements in model
-                model = handlers.substitute(self.model, params)
-                return constrain_fn(model, self._inv_transforms, model_args,
-                                    model_kwargs, unpacked_samples)
-            else:
-                return transform_fn(self._inv_transforms, unpacked_samples)
-
-        if sample_shape:
-            latent_sample = np.reshape(latent_sample, (-1, np.shape(latent_sample)[-1]))
-            unpacked_samples = vmap(unpack_single_latent)(latent_sample)
-            unpacked_samples = tree_map(lambda x: np.reshape(x, sample_shape + np.shape(x)[1:]),
-                                        unpacked_samples)
-        else:
-            unpacked_samples = unpack_single_latent(latent_sample)
-        return unpacked_samples
+    def _unpack_and_constrain(self, latent_sample):
+        return transform_fn(self._inv_transforms, self._unpack_latent(latent_sample))
```

Comment:
We save a bunch of code here. :D

Reply:
@neerajprad I'll add a commit to record `deterministic` sites for autoguide. I think we did it previously here.